src/lib/expansion/expansion-panel-header.ts

```diff
@@ -114,6 +127,28 @@ export class MatExpansionPanelHeader implements OnDestroy, FocusableOption {
   /** Height of the header while the panel is collapsed. */
   @Input() collapsedHeight: string;
 
+  /**
+   * Role for the header. By default, the role is "button". If the header is not considered
+   * to be a button, e.g. when the header contains nested buttons, the role should not be
+   * set to "button". Notice that in this case the toggle button will have a role of "button".
+   */
+  @Input() headerRole: MatExpansionPanelHeaderRole = 'button';
```

Comment:
Rather than exposing an API for an arbitrary header `role`, I'd lean more towards having it be a boolean flag that toggles between `button` and `presentation`. Maybe something like `removeButtonRole`.

Another option, instead of adding a new `@Input`, we could simply check for the presence of a `role` attribute on the header element to remove all of the other `aria-` attributes from the header and move them to the toggle indicator.

Yet another option, we could add an input like
```ts
@Input()
toggleElement = 'header' | 'indicator' | 'custom';
```

@crisbeto @josephperrott opinions here?

Reply:
@jelbourn Any further thoughts on I should proceed? I think the easiest way would be change the `MatExpansionPanelHeaderRole` type to be:

```
/** Role for MatExpansionPanelHeader. */
export type MatExpansionPanelHeaderRole = 'button' | 'presentation' | null;
```

but let me know if you have any concerns.